src/packagedcode/pypi.py

```diff
@@ -274,12 +276,37 @@ def parse_with_dparse(location):
 
 def parse_requirements_txt(location):
     """
-    Return a package built from requirements.txt.
+    Return a package built from Python requirements.txt files.
     """
     package_dependencies = parse_with_dparse(location)
     return PythonPackage(dependencies=package_dependencies)
 
 
+def parse_pipfile_lock(location):
+    """
+    Return package built from Python Pipfile.lock files.
+    """
+    with open(location, 'r') as f:
+        content = f.read()
+
+    try:
+        data = json.loads(content, object_pairs_hook=OrderedDict)
+    except json.decoder.JSONDecodeError:
+        data = {}
+
+    sha256 = None
+    if '_meta' in data:
+        for name, meta in data['_meta'].items():
+            if name=='hash':
+                sha256 = meta.get('sha256')
```

Comment:
Do you know what is this sha256 about? A download archive? the original Pipfile?

Reply:
@pombredanne This was originally brought up by me. It the the sha256 of the original pipfile (I think). Since it was in the data, so I suggested we should collect it. We may not want to do this, as it is slightly confusing. Also, I am unsure if this particular hash will be present in most pipfiles.

Reply:
@pombredanne The sha256 which we are collecting in this packagedcode is of original pipfile. So I think we should detect this and add this in sha256 :)

Reply:
ok, please enter a ticket to revisit this. I am not sure we can do much of anything with this as this is not the sha256 of some package archive of sorts?